providers/openfeature-go-feature-flag-provider/lib/openfeature/go-feature-flag/goff_api.rb

```diff
@@ -0,0 +1,164 @@
+        response = http.request(request)
+
+        case response.code.to_i
+        when 200
+          parse_success_response(response)
+        when 400
+          parse_error_response(response)
+        when 401, 403
+          raise OpenFeature::GoFeatureFlag::UnauthorizedError.new(response)
+        when 404
+          raise OpenFeature::GoFeatureFlag::FlagNotFoundError.new(response, flag_key)
+        when 429
+          parse_retry_later_header(response)
+          raise OpenFeature::GoFeatureFlag::RateLimited.new(response)
+        else
+          raise OpenFeature::GoFeatureFlag::InternalServerError.new(response)
+        end
+      end
+
+      private
+
+      def parse_error_response(response)
+        required_keys = %w[key error_code]
+        parsed = JSON.parse(response.body)
+
+        missing_keys = required_keys - parsed.keys
+        unless missing_keys.empty?
+          raise OpenFeature::GoFeatureFlag::ParseError.new(response)
+        end
+
+        OpenFeature::GoFeatureFlag::OfrepApiResponse.new(
+          value: nil,
+          key: parsed["key"],
+          reason: OpenFeature::SDK::Provider::Reason::ERROR,
+          variant: nil,
+          error_code: error_code_mapper(parsed["error_code"]),
+          error_details: parsed["error_details"],
+          metadata: nil
+        )
+      end
+
+      def parse_success_response(response)
+        required_keys = %w[key value reason variant]
+        parsed = JSON.parse(response.body)
+
+        missing_keys = required_keys - parsed.keys
+        unless missing_keys.empty?
+          raise OpenFeature::GoFeatureFlag::ParseError.new(response)
+        end
+
+        OpenFeature::GoFeatureFlag::OfrepApiResponse.new(
+          value: parsed["value"],
+          key: parsed["key"],
+          reason: reason_mapper(parsed["reason"]),
+          variant: parsed["variant"],
+          error_code: nil,
+          error_details: nil,
+          metadata: parsed["metadata"]
+        )
+      end
+
+      def reason_mapper(reason_str)
+        reason_str = reason_str.upcase
+        reason_map = {
+          "STATIC" => OpenFeature::SDK::Provider::Reason::STATIC,
+          "DEFAULT" => OpenFeature::SDK::Provider::Reason::DEFAULT,
+          "TARGETING_MATCH" => OpenFeature::SDK::Provider::Reason::TARGETING_MATCH,
+          "SPLIT" => OpenFeature::SDK::Provider::Reason::SPLIT,
+          "CACHED" => OpenFeature::SDK::Provider::Reason::CACHED,
+          "DISABLED" => OpenFeature::SDK::Provider::Reason::DISABLED,
+          "UNKNOWN" => OpenFeature::SDK::Provider::Reason::UNKNOWN,
+          "STALE" => OpenFeature::SDK::Provider::Reason::STALE,
+          "ERROR" => OpenFeature::SDK::Provider::Reason::ERROR
+        }
+        reason_map[reason_str] || OpenFeature::SDK::Provider::Reason::UNKNOWN
+      end
+
+      def error_code_mapper(error_code_str)
+        error_code_str = error_code_str.upcase
+        error_code_map = {
```

Comment:
This does not include `PROVIDER_FATAL`, is it on purpose because this is not something goff could ever return?

Reply:
Exactly, `PROVIDER_FATAL` will never come from goff since this is something that can happen only inside the provider.